doc/source/reference/array_api.rst

```diff
@@ -0,0 +1,795 @@
+   * - ``round``
+     - Numeric
+   * - ``sign``
+     - Numeric
+   * - ``sin``
+     - Floating-point
+   * - ``sinh``
+     - Floating-point
+   * - ``sqrt``
+     - Floating-point
+   * - ``square``
+     - Numeric
+   * - ``subtract``
+     - Numeric
+   * - ``tan``
+     - Floating-point
+   * - ``tanh``
+     - Floating-point
+   * - ``trunc``
+     - Numeric
+
+(*) These functions have different names from the main ``numpy`` namespace.
+See :ref:`array_api-name-changes`.
+
+Creation Functions
+~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+
+   * - Function Name
+     - Dtypes
+   * - ``meshgrid``
+     - Any (all input dtypes must be the same)
+
+
+Linear Algebra Functions
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+   :header-rows: 1
+
+   * - Function Name
+     - Dtypes
+   * - ``cholesky``
+     - Floating-point
+   * - ``cross``
+     - Numeric
+   * - ``det``
+     - Floating-point
+   * - ``diagonal``
+     - Any
+   * - ``eigh``
+     - Floating-point
+   * - ``eighvals``
+     - Floating-point
+   * - ``inv``
+     - Floating-point
+   * - ``matmul``
+     - Numeric
+   * - ``matrix_norm`` (*)
+     - Floating-point
+   * - ``matrix_power``
+     - Floating-point
+   * - ``matrix_rank``
+     - Floating-point
+   * - ``matrix_transpose`` (**)
+     - Any
+   * - ``outer``
+     - Numeric
+   * - ``pinv``
+     - Floating-point
+   * - ``qr``
+     - Floating-point
+   * - ``slogdet``
+     - Floating-point
+   * - ``solve``
+     - Floating-point
+   * - ``svd``
+     - Floating-point
+   * - ``svdvals`` (**)
+     - Floating-point
+   * - ``tensordot``
+     - Numeric
+   * - ``trace``
+     - Numeric
+   * - ``vecdot`` (**)
+     - Numeric
+   * - ``vector_norm`` (*)
+     - Floating-point
+
+(*) Thes functions are split from ``norm`` from the main ``numpy`` namespace.
+See :ref:`array_api-name-changes`.
+
+(**) These functions are new in the array API and are not in the main
+``numpy`` namespace.
+
+Array Object
+~~~~~~~~~~~~
+
+All the special ``__operator__`` methods on the array object behave
+identically to their corresponding functions (see `the spec
+<https://data-apis.org/array-api/latest/API_specification/array_object.html#methods>`__
+for a list of which methods correspond to which functions). The exception is
+that operators explicitly allow Python scalars according to the `rules
+outlined in the spec
+<https://data-apis.org/array-api/latest/API_specification/type_promotion.html#mixing-arrays-with-python-scalars>`__
+(see :ref:`array_api-type-promotion-differences`).
+
+
+Array Object Differences
+------------------------
+
+.. list-table::
+   :header-rows: 1
+
+   * - Feature
+     - Type
+     - Notes
+   * - No array scalars
+     - ???
+     - The spec does not have array scalars, only 0-D arrays. It is not clear
+       if NumPy's scalars deviate from the spec 0-D array behavior in any ways
+       other than the ones outlined in
+       :ref:`array_api-type-promotion-differences`.
```

Comment:
Right, I wasn't clear if there were any other odd differences. Immutability shouldn't be a problem because strictly speaking the spec allows it (some libraries like Jax are completely immutable, c.f. https://data-apis.org/array-api/latest/design_topics/copies_views_and_mutation.html).

Reply:
I changed this to "strictness" for now. I believe that other than type promotion (which is discussed elsewhere), scalars can duck type as 0-D arrays sufficiently for the spec. If this is ever disproved, we can update this. It's hard to know for sure without plugging them into the test suite somehow, unless you are aware of any other differences that could affect things.